src/graph/test/GroupByLimitTest.cpp

```diff
@@ -0,0 +1,237 @@
+        std::vector<std::tuple<std::string, double, uint64_t>> expected = {
+                {"LeBron James", 3.0, 2},
+                {"Chris Paul", 3.0, 2},
+                {"Dwyane Wade", 1.5, 1},
+                {"Carmelo Anthony", 1.5, 1},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+
+
+    // output next
+    {
+        cpp2::ExecutionResponse resp;
+        auto &player = players_["Paul Gasol"];
+        auto *fmt = "GO FROM %ld OVER like "
+                    "YIELD $$.player.age as age, "
+                    "like._dst as id "
+                    "| GROUP BY $-.id "
+                    "YIELD $-.id as id, "
+                    "SUM(age) as age |"
+                    "GO FROM $-.id OVER serve "
+                    "YIELD $$.team.name as name";
+        auto query = folly::stringPrintf(fmt, player.vid());
+        auto code = client_->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        std::vector<std::tuple<std::string>> expected = {
+                {"Grizzlies"},
+                {"Raptors"},
+                {"Lakers"},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+
+    // group input nonexistent
+    {
+        cpp2::ExecutionResponse resp;
+        auto &player = players_["Marco Belinelli"];
+        auto *fmt = "GO FROM %ld OVER serve "
+                    "YIELD $$.team.name as name, "
+                    "serve._dst as id"
+                    "| GROUP BY $-.start_year "
+                    "YIELD COUNT($-.id)";
+        auto query = folly::stringPrintf(fmt, player.vid());
+        auto code = client_->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code);
+    }
+
+    // group alias nonexistent
+    {
+        cpp2::ExecutionResponse resp;
+        auto &player = players_["Marco Belinelli"];
+        auto *fmt = "GO FROM %ld OVER serve "
+                    "YIELD $$.team.name as name, "
+                    "serve._dst as id"
+                    "| GROUP BY team "
+                    "YIELD COUNT($-.id), "
+                    "$-.name as teamName";
+        auto query = folly::stringPrintf(fmt, player.vid());
+        auto code = client_->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code);
+    }
+
+    // field nonexistent
+    {
+        cpp2::ExecutionResponse resp;
+        auto &player = players_["Marco Belinelli"];
+        auto *fmt = "GO FROM %ld OVER serve "
+                    "YIELD $$.team.name as name, "
+                    "serve._dst as id"
+                    "| GROUP BY $-.name "
+                    "YIELD COUNT($-.start_year)";
+        auto query = folly::stringPrintf(fmt, player.vid());
+        auto code = client_->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code);
+    }
+}
+
+TEST_F(GroupByLimitTest, GroupByLimitTest) {
+}
+
+TEST_F(GroupByLimitTest, GroupByOrderByLimitTest) {
+}
+}  // namespace graph
```

Comment:
so you will add some case including 
group by, order by, and limit
would you?

Reply:
yeah